packages/permissionless/accounts/trust/signerToTrustSmartAccount.ts

```diff
@@ -0,0 +1,234 @@
+>(
+    signer: SmartAccountSigner<TSource, TAddress>,
+    chainId: number,
+    accountAddress: Address,
+    hashedMessage: Hex
+): Promise<Hex> {
+    return signer.signTypedData({
+        domain: {
+            chainId: Number(chainId),
+            name: "Barz",
+            verifyingContract: accountAddress,
+            version: "v0.2.0"
+        },
+        types: {
+            BarzMessage: [{ name: "message", type: "bytes" }]
+        },
+        message: {
+            message: hashedMessage
+        },
+        primaryType: "BarzMessage"
+    })
+}
+
+/**
+ * Default addresses for Trust Smart Account
+ */
+export const TRUST_ADDRESSES: {
+    Secp256k1VerificationFacetAddress: Address
+    factoryAddress: Address
+} = {
+    Secp256k1VerificationFacetAddress:
+        "0x81b9E3689390C7e74cF526594A105Dea21a8cdD5",
+    factoryAddress: "0x729c310186a57833f622630a16d13f710b83272a"
+}
+
+export type TrustSmartAccount<
+    entryPoint extends EntryPoint,
+    transport extends Transport = Transport,
+    chain extends Chain | undefined = Chain | undefined
+> = SmartAccount<entryPoint, "TrustSmartAccount", transport, chain>
+
+export type SignerToTrustSmartAccountParameters<
+    entryPoint extends EntryPoint,
+    TSource extends string = string,
+    TAddress extends Address = Address
+> = {
+    signer: SmartAccountSigner<TSource, TAddress>
+    factoryAddress?: Address
+    entryPoint: entryPoint
+    index?: bigint
+    address?: Address
+    secp256k1VerificationFacetAddress?: Address
+}
+
+/**
+ * @description Creates an Trust Smart Account from a private key.
+ *
+ * @returns A Private Key Trust Smart Account.
+ */
+export async function signerToTrustSmartAccount<
+    entryPoint extends EntryPoint,
+    TTransport extends Transport = Transport,
+    TChain extends Chain | undefined = Chain | undefined,
+    TSource extends string = string,
+    TAddress extends Address = Address
+>(
+    client: Client<TTransport, TChain, undefined>,
+    {
+        signer,
+        factoryAddress = TRUST_ADDRESSES.factoryAddress,
+        entryPoint: entryPointAddress,
+        index = 0n,
+        secp256k1VerificationFacetAddress = TRUST_ADDRESSES.Secp256k1VerificationFacetAddress,
+        address
+    }: SignerToTrustSmartAccountParameters<entryPoint, TSource, TAddress>
+): Promise<TrustSmartAccount<entryPoint, TTransport, TChain>> {
+    const viemSigner: LocalAccount<string> = {
+        ...signer,
+        signTransaction: (_, __) => {
+            throw new SignTransactionNotSupportedBySmartAccount()
+        }
+    }
+
+    const [accountAddress, chainId] = await Promise.all([
+        address ??
+            getAccountAddress(client, {
+                factoryAddress,
+                secp256k1VerificationFacetAddress,
+                entryPoint: entryPointAddress,
+                bytes: viemSigner.publicKey,
+                owner: viemSigner.address,
+                index
+            }),
+        getChainId(client)
```

Comment:
Can we check if client has the chain so

`client.chain?.id ?? getChainId(client)`

Reply:
Done